pkg/gatherers/clusterconfig/install_plans_test.go

```diff
@@ -58,45 +58,47 @@ func Test_InstallPlans_Gather(t *testing.T) {
 			var client *dynamicfake.FakeDynamicClient
 			coreClient := kubefake.NewSimpleClientset()
 			for _, file := range test.testfiles {
-				f, err := os.Open(file)
-				if err != nil {
-					t.Fatal("test failed to read installplan data", err)
-				}
-				defer f.Close()
-				installplancontent, err := io.ReadAll(f)
-				if err != nil {
-					t.Fatal("error reading test data file", err)
-				}
+				func() {
+					f, err := os.Open(file)
+					if err != nil {
+						t.Fatal("test failed to read installplan data", err)
+					}
+					defer f.Close()
```

Comment:
This defer is still in a 'loop' (which I thought this PR was removing)? Or are we only removing it in 'some places'?